### PR TITLE
Finished both maps for the dashboard : Caleb

### LINKED ISSUE #255: Make a map diagram for where each body is visible
As an astronomer I would like to see where in the UK a given body is visible on a night, so I want to be able to visualise this so I can travel to different spots to stargaze

- Add a visual that shows an overview of how visible each body is.

### LINKED ISSUE #273: Make a map diagram showing the visibility of each county
As an astronomer I would like to see what the visibility and cloud coverage so I have a rough idea of what areas stargazing will be possible that night.

Add a visual that shows an overview average visibility and cloud coverage.

### PR DESCRIPTION
**Pull request by CalebMathers**
This closes #255 
This closes #273 

Wrote function to produce two more maps;
- One that, given a celestial body, shows which regions it will be visible from that night, along with the average cloud coverage for the same period.
- Another that shows the average visibility and cloud coverage for the next night per county.

### CODE REVIEW

File: dashboard/aurora_map.py [hunk 1]
```diff
@@ -208,19 +215,213 @@ def create_aurora_map() -> plt.figure:
...
+            cur.execute(query)
+            result = cur.fetchall()
+
+    return {row['county_name']: float(row['avg_cloud_cov']) for row in result}
+
+
+def get_body_visible_regions(body: str) -> list[str]:
+    '''Returns a list of regions that will have a given body appear at some
+    point over the night period 1700-0500'''
+    query = """SELECT DISTINCT r.region_name
```
Comment (AlexOBarnes) [inline dashboard/aurora_map.py:258]: Lovely query

File: dashboard/aurora_map.py [hunk 2]
```diff
@@ -208,19 +215,213 @@ def create_aurora_map() -> plt.figure:
...
     return fig
 
 
-def create_visibilty_map() -> plt.figure:
-    '''Returns a map figure.'''
-    pass
+def get_average_visibility_data() -> dict:
+    '''Returns a dict of all counties average visibility 
+    in meters for the night 1700-0500.'''
+    query = """SELECT c.county_name,
```
Comment (AlexOBarnes) [inline dashboard/aurora_map.py:342]: Lovely query

File: dashboard/aurora_map.py [hunk 3]
```diff
@@ -208,19 +215,213 @@ def create_aurora_map() -> plt.figure:
...
+                continue
+            if avg_visibility[county] < 40000:
+                mapped_values[area] = 'purple'
+                continue
+            mapped_values[area] = 'gold'
+
+    return mapped_values
+
+
+def create_visibility_map() -> plt.Figure:
```
Comment (AlexOBarnes) [inline dashboard/aurora_map.py:385]: Thanks for spelling this correctly

**Review by AlexOBarnes — APPROVED** [review]
Looks good to me

File: dashboard/aurora_map.py [hunk 4]
```diff
@@ -1,16 +1,21 @@
+'''File to generate maps for the dashboard.'''
+
+from os import environ as ENV
+
```
Comment (olliejth) [inline dashboard/aurora_map.py:4]: Good clean imports.

File: dashboard/aurora_map.py [hunk 5]
```diff
@@ -172,34 +177,36 @@ def map_cloud_coverage() -> dict:
...
-            key: 'yellow' for key in map_dict if key not in NORTHERN_REGIONS}
+            region: 'blue' for region in map_regions if region not in NORTHERN_REGIONS}
         for region in NORTHERN_REGIONS:
-            colours[region] = 'orange'
+            colours[region] = 'green'
     else:
-        colours = {key: 'orange' for key in map_dict}
+        colours = {region: 'green' for region in map_regions}
 
     return colours
```
Comment (olliejth) [inline dashboard/aurora_map.py:199]: Good reversal of colours.

File: dashboard/aurora_map.py [hunk 6]
```diff
@@ -172,34 +177,36 @@ def map_cloud_coverage() -> dict:
...
 
 
 def create_aurora_map() -> plt.figure:
-    '''Returns a map figure.'''
-    fig, ax = plt.subplots()
```
Comment (olliejth) [inline dashboard/aurora_map.py:199]: Could be a more descriptive docstring.

File: dashboard/aurora_map.py [hunk 7]
```diff
@@ -172,34 +177,36 @@ def map_cloud_coverage() -> dict:
...
+    '''Returns a map figure showing the possibility of an aurora sighting per region
+    for the current time, over-layed with a map of each counties current cloud
+    coverage as a white alpha value.'''
+    fig, ax = plt.subplots(figsize=(5, 5))
```
Comment (olliejth) [inline dashboard/aurora_map.py:207]: Much better.

File: dashboard/aurora_map.py [hunk 8]
```diff
@@ -208,19 +215,213 @@ def create_aurora_map() -> plt.figure:
...
+    current cloud coverage.'''
+
+    props = {'boxstyle': 'round', 'facecolor': 'white', 'alpha': 0.5}
+    ax.text(1.6, 0.8, legend_text, fontsize=6, bbox=props,
+            transform=ax.transAxes, verticalalignment='top', horizontalalignment='right')
+
     ax.set_axis_off()
     plt.gcf().set_facecolor('black')
 
     return fig
```
Comment (olliejth) [inline dashboard/aurora_map.py:232]: Oversimplification I feel.

**Review by olliejth — APPROVED** [review]
Agree with the changes.

**Pull request merged**